Add support to explicitly disable ENA support

bodgit opened the pull request:
If `ena_support` is set to false then it previously didn't do anything
whereas now it will explicitly disable ENA support.

Fixes #6852

azr commented:
Nice one ! Won't this one need some documentation ?

SwampDragons commented:
@bodgit thanks for doing this!  I agree with @azr -- this looks great except I'd like to see a note in the docs clarifying the new behavior.  

https://github.com/hashicorp/packer/tree/master/website/source/docs/builders/amazon-chroot.html.md
https://github.com/hashicorp/packer/tree/master/website/source/docs/builders/amazon-ebs.html.md
https://github.com/hashicorp/packer/tree/master/website/source/docs/builders/amazon-ebssurrogate.html.md
https://github.com/hashicorp/packer/tree/master/website/source/docs/builders/amazon-ebsvolume.html.md
https://github.com/hashicorp/packer/tree/master/website/source/docs/builders/amazon-instance.html.md

bodgit commented:
Fixed. I removed the note about optionally requiring the `ec2:ModifyInstanceAttribute` permission as it's effectively mandatory now, but it's also listed in the minimal IAM policy for Packer on AWS anyway.

Ok?

azr commented on builder/amazon/common/step_modify_ebs_instance.go:
```diff
@@ -37,20 +38,24 @@ func (s *StepModifyEBSBackedInstance) Run(_ context.Context, state multistep.Sta
+		prefix = "En"
+	} else {
+		prefix = "Dis"
+	}
```
I was just thinking, what if `s.EnableAMIENASupport` was a pointer; then if it's `nil` then we keep whatever setting we had.

bodgit replied:
That would make sense as it seemed to me that the setting was either `true` or `false`, there was no undefined state. Would that be preferred?

SwampDragons replied:
My understanding is that this PR's change forces true vs false rather than having an undefined state, which technically changes the behavior; prior to this change the options were essentially "true" or "undefined", or you wouldn't have had the issues you saw.  Using @azr's suggestion would allow us to maintain backwards behavior where not setting the value means that we won't change it either way.

bodgit replied:
Ok, I will see if I can rework this to be a pointer

bodgit commented:
I've now reworked `ena_support` to be of type `*bool`. I've tested three Packer builds; one with no `ena_support` parameter set and one each with `ena_support` set to `true` and `false`. All three pass, and the latter two only mention enabling/disabling ENA support.

bodgit commented:
Docs updated again to match behaviour.
